src/offset/local/tz_info/timezone.rs

```diff
@@ -26,9 +27,9 @@ impl TimeZone {
     ///
     /// This method in not supported on non-UNIX platforms, and returns the UTC time zone instead.
     ///
-    pub(crate) fn local() -> Result<Self, Error> {
-        if let Ok(tz) = std::env::var("TZ") {
-            Self::from_posix_tz(&tz)
+    pub(crate) fn local(env_result: &Result<String, VarError>) -> Result<Self, Error> {
```

Comment:
this may be better as `&Option<String>` or similar

Reply:
So this is in order to reuse the result for both `Source::new()` and `TimeZone::local()`, right? I think we should probably pass it as `Option<&str>`. Also this change might make sense to do as a separate commit?

Reply:
excellent, I've gone for `Option<&str>` in [d2a32c8](https://github.com/chronotope/chrono/pull/853/commits/d2a32c8e50a44fdfd5f33bc4fcc11b12da5507eb)